friend/controller.js

```diff
@@ -0,0 +1,72 @@
+const router = require('express').Router();
+const Friends = require('./friends.js');
+
+router
+    .route('/')
+    .get((req, res) => {
+        Friends.find()
+        .then( friends => {
+            res.status(200).json(friends)
+        
+        })
+        .catch( err => {
+            res.status(500).json({error: "error getting friends"})
+        })
+    })
+    .post((req, res) => {
+        const { body } = req;
+     
+//validates scheme with the New Property
+        const friends = new Friends(req.body);
+        //   
+        friends.save()
+        .then(friends => {
+            res.status(200).json(friends)
+        })
+        .catch(err => {
+            res.status(500).json(err)
+        })
+    })
+    
+router
+    .route('/:id')
+    .get((req, res) => {
+    
+        res.status(200).json({route: '/api/friends/' + req.params.id})
```

Comment:
We could use Mongoose's `findById` (http://mongoosejs.com/docs/api.html#findbyid_findById) method here to find our friend:

```
.get((req, res) => {
  Friends.findById(req.params.id)
  .then(response => {
    if (response) res.status(200).json(response)
    else res.status(404).json({ message: "The friend with the specified ID does not exist." })
  })
  .catch(err => res.status(500).json({ errorMessage: "The friend information could not be retrieved." }))
})
```